pkg/providers/provider.go

```diff
@@ -26,6 +26,7 @@ type Provider interface {
 	PostWorkloadInit(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error
 	BootstrapClusterOpts(clusterSpec *cluster.Spec) ([]bootstrapper.BootstrapClusterOption, error)
 	UpdateKubeConfig(content *[]byte, clusterName string) error
+	AdditionalFiles() map[string][]byte
```

Comment:
What is this being used for? Sorry I can't find it
I'm just concerned about adding even more stuff to this interface, specially when is only used by one provider

Reply:
Sorry, the usage isn't in this PR. I split it off to keep the PR focused on generating the hardware yaml. 

For context, this was originally`HardwareSpec()`, but we decided to make it AdditionalFiles instead so it would make more sense being on the provider interface. It is going to be used to retrieve the generated hardwareSpec in the CLI workflow during to upload it to GitOps like so:

```
func (s *InstallGitOpsManagerTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
...
        var hardwareSpec []byte
        if commandContext.Provider.AdditionalFiles != nil {
            hardwareSpec = commandContext.Provider.AdditionalFiles["hardware"]
        }
        
	err := commandContext.GitOpsManager.InstallGitOps(..., ..., ..., ..., ..., hardwareSpec)
...
}
```